numba_cuda/numba/cuda/cudadrv/nvrtc.py

```diff
@@ -251,6 +254,31 @@ def compile(src, name, cc, ltoir=False):
     nvrtc = NVRTC()
     program = nvrtc.create_program(src, name)
 
+    version = nvrtc.get_version()
+    if version < (11, 0):
+        # Maximum supported target architecture for 11.x is cc 9.0 in CUDA 11.8
+        raise RuntimeError(
+            "Unsupported CUDA version. CUDA 11.0 or higher is required."
+        )
+    else:
+        min_cc, max_cc = CTK_SUPPORTED[version]
+        if version < min_cc:
+            raise RuntimeError(
+                f"Device Compute Capability ({cc}) is lower "
+                f"than minimum supported CC ({min_cc}) with installed "
+                "cudatoolkit. Consider downgrading cudatoolkit to supported "
+                "version."
+            )
+        elif version > max_cc:
+            warnings.warn(
+                f"Device Compute Capability ({cc}) is higher "
+                f"than maximum supported CC ({max_cc}) with installed "
+                "cudatoolkit. Numba will limit the compilation to the maximum"
+                "supported CC. This may hinder performance, consider upgrading"
+                "your cudatoolkit to supported version."
+            )
```

Comment:
Similar to the above. In this case, I think upgrading NVRTC is an appropriate recommendation in the warning.
```suggestion
            warnings.warn(
                f"Device Compute Capability ({cc}) is higher than the "
                f"maximum supported Compute Capability ({max_cc}) with "
                f"NVRTC {".".join(version)}. Numba will limit the compilation "
                f"to the maximum supported Compute Capability ({max_cc}). "
                "This may hinder performance, consider upgrading NVRTC."
            )
```